### PR TITLE
[Perf] Make latest_output_ids H2D non-blocking in prepare_for_decode

### PR DESCRIPTION
**Pull request by hnyls2002**
`prepare_for_decode` builds `latest_output_ids` with a blocking `torch.tensor(..., device=...)`, unlike the other per-step H2D copies in this path which all use `non_blocking=True`. Under the overlap scheduler this forces a `cudaStreamSynchronize` every decode step that stalls the CPU behind the in-flight forward — most visible at small batch / low latency.

`forward_stream` already waits on `schedule_stream`, so the async copy stays ordered before the forward consumes it. For penalties that don't read the tensor (e.g. `min_new_tokens`, which only counts) it was pure overhead.

## Benchmark

`openai/gpt-oss-120b`, TP=4 (GB300), overlap scheduler. Workload is `aiperf` 1k-in / 1k-out at concurrency 1 (bs=1 decode), with `min_tokens` set so the penalizer path that builds `latest_output_ids` is active. `fwd occupancy` = median of `sglang:fwd_occupancy` from `/metrics` while the bench runs.

Launch:

```
SGLANG_ENABLE_METRICS_DEVICE_TIMER=1 python -m sglang.launch_server \
  --model openai/gpt-oss-120b --tp 4 --enable-metrics \
  --cuda-graph-bs-decode 1 2 4 8
```

Bench:

```
aiperf profile \
  --model openai/gpt-oss-120b --url http://127.0.0.1:30000 \
  --endpoint-type chat --streaming \
  --isl 1024 --osl 1024 --concurrency 1 \
  --extra-inputs min_tokens:1024 --extra-inputs ignore_eos:true \
  --benchmark-duration 85
```

|                          | fwd occupancy | decode throughput | inter-token latency |
| ------------------------ | ------------- | ----------------- | ------------------- |
| before (blocking H2D)    | 71.4%         | 255 tok/s         | 3.91 ms             |
| after (non-blocking H2D) | 86.9%         | 330 tok/s         | 2.96 ms             |

When no penalizer is active the copy is skipped entirely, so that path is unchanged (~90% occupancy either way).

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #27666195284](https://github.com/sgl-project/sglang/actions/runs/27666195284)
Latest PR Test (Extra): :x: [Run #27666195213](https://github.com/sgl-project/sglang/actions/runs/27666195213)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test registered/sampling/test_penalty.py registered/core/test_srt_endpoint.py
